src/charm.py

```diff
@@ -156,13 +156,18 @@ def _on_install(self, event: InstallEvent) -> None:
         if self.unit.is_leader():
             self.state.cluster.update({"quorum": "default - non-ssl"})
 
-    def _on_cluster_relation_changed(self, event: EventBase) -> None:
+    def _on_cluster_relation_changed(self, event: EventBase) -> None:  # noqa: C901
```

Comment:
The new condition brings us past the quality threshold for complexity in this method. Is it ok to ignore the rule here? I believe so since this is the "something change, handle it" event handler.

Reply:
I believe that's all right, also because the new addition is not really adding any complexity, but the linting is probably right on saying that maybe we should try to gather logically parts of the code into higher level functions. 

But this is clearly outside of the scope here, given that really - the modification here is very tiny with respect to the business logic